vespalib/src/vespa/vespalib/hwaccelrated/private_helpers.hpp

```diff
@@ -74,5 +74,31 @@ orChunks(size_t offset, const std::vector<std::pair<const void *, bool>> & src,
     }
 }
 
+template<typename TemporaryT=int32_t>
+double euclideanDistanceT(const int8_t * a, const int8_t * b, size_t sz) __attribute__((noinline));
+template<typename TemporaryT>
+double euclideanDistanceT(const int8_t * a, const int8_t * b, size_t sz)
+{
+    //Note that this is 3 times faster with int32_t than with int64_t and 16x faster than float
+    TemporaryT sum = 0;
+    for (size_t i(0); i < sz; i++) {
+        int16_t d = int16_t(a[i]) - int16_t(b[i]);
+        sum += d * d;
+    }
+    return sum;
+}
+
+inline double
+euclideanDistance(const int8_t * a, const int8_t * b, size_t sz) {
```

Comment:
should use 'squared' in the name here as well

Reply:
Done